iceoryx_utils/include/iceoryx_utils/posix_wrapper/timer.hpp

```diff
@@ -321,6 +321,58 @@ class Timer
 };
 
 } // namespace posix
+
+
+namespace clock
+{
+/// @brief This offers the deadline timer functionality. It has user convenient methods to reset the timer [by default
+/// it uses the intialized duration], reset timer to a customized duration, check if the timer is active and user can
+/// also get to know about the remaining time before the timer goes off
+/// @note Introduced as an alternate timer functionality for the deprecated POSIX timers
+///
+/// @code
+///     iox::clock::DeadlineTimer deadlineTimer(1000_ms);
+///
+///     // to check if the timer is active
+///     if( deadlineTimer.hasExpired()){
+///     ...
+///     }
+///     // to reset the timer and start again with the same duration
+///     deadlineTimer.reset();
+///
+/// @endcode
+class DeadlineTimer
+{
+  public:
+    /// @brief Constructor
+    /// @param[in] timeToWait duration until the timer expires
+    DeadlineTimer(const iox::units::Duration timeToWait) noexcept;
+
+    /// @brief Checks if the timer has expired compared to its absolute end time
+    /// @return true if the timer is still active and false if it is expired
+    bool hasExpired() noexcept;
+
+    /// @brief reinitializes the ending time for the timer. The absolute end time is calculated by adding time to wait
+    /// to the current time.
+    void reset() noexcept;
+
+    /// @brief reinitializes the ending time for the timer to the given new time to wait. The absolute end time is
+    /// calculated by adding new time to wait to the current time.
+    void reset(const iox::units::Duration timeToWait) noexcept;
```

Comment:
please add a `@param[in]`

Reply:
forget the `@brief` part, the line was just wrapped on my monitor and I looked like there was an empty line above `/// calculate`

Reply:
Okay added `@param[in]`